pkg/middleware/csrf.go

```diff
@@ -34,16 +35,30 @@ func CSRF(loginCookieName string, logger log.Logger) func(http.Handler) http.Han
 				http.Error(w, err.Error(), http.StatusBadRequest)
 				return
 			}
-
 			origin, err := url.Parse(r.Header.Get("Origin"))
 			if err != nil {
 				logger.Error("error parsing Origin header", "err", err)
 			}
-			if err != nil || netAddr.Host == "" || (origin.String() != "" && origin.Hostname() != netAddr.Host) {
-				http.Error(w, "origin not allowed", http.StatusForbidden)
-				return
+			// env var GF_USE_BEHIND_PROXY controls if X-Forwarded-Host is used or not
+			if UseBehindProxy := os.Getenv("GF_USE_BEHIND_PROXY"); UseBehindProxy == "true" {
```

Comment:
Instead of insisting on the use of `X-Forwarded-Host` would it not be better to have a variable such as 
`GF_PROXY_FORWARDED_HOST_HEADER="X-Forwarded-Host`

Reply:
I don't think this should be configured through an environment variable at all, but with an option in grafana.ini like all the other settings too.

Reply:
Using the `grafana.ini` is of course also perfectly fine for me. I've added a note to the initial PR description that I've used the env var as this was the easiest solution for me to develop a workaround:

> I've added an additional env var GF_USED_BEHIND_PROXY that allows us to enable (true) or disable (false or empty) the check of X-Forwarded-Host when not needed. Its using os.Getenv - I saw that there's a builtin mechanism to leverage env vars but I haven't figured out how to use it. This as well as the name of the env var might have to change.****

It would be great if you could help me to change it from using envvars to `grafana.ini` in case that increases the chances to get it into grafana at some point.